src/pages/workspaces/workspace/Dashboard.js

```diff
@@ -1,38 +1,28 @@
-import _ from 'lodash'
-import { Component } from 'react'
 import { div, hh } from 'react-hyperscript-helpers/lib/index'
+import { buttonPrimary } from 'src/components/common'
 import { spinner } from 'src/components/icons'
+import Modal from 'src/components/Modal'
 import * as Ajax from 'src/libs/ajax'
 import * as Style from 'src/libs/style'
-import { buttonPrimary } from 'src/components/common'
-import Modal from 'src/components/Modal'
+import { Component } from 'src/libs/wrapped-components'
+import * as Utils from 'src/libs/utils'
 
 
 export default hh(class WorkspaceDashboard extends Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      workspace: null
-    }
-  }
-
-
   componentWillMount() {
     const { namespace, name } = this.props
     Ajax.workspace.details(namespace, name,
       workspace => this.setState({ workspace }),
       failure => this.setState({ failure }))
   }
 
-
   render() {
     const { workspace, failure, modal } = this.state
 
-    return _.isEmpty(workspace) ?
-      failure ?
-        `Couldn't load workspace details: ${failure}` :
-        spinner({ style: { marginTop: '1rem' } }) :
-      div({ style: { margin: '1rem' } }, [
+    return Utils.cond(
+      [failure, `Couldn't load workspace details: ${failure}`],
+      [!workspace, spinner({ style: { marginTop: '1rem' } })],
```

Comment:
I consider the `render` method in React to be similar to a tight loop in performance critical code. Despite the lack of elegance, I'd prefer the values to always be functions so only the truthy case is executed.

I do like the choice of using lists to differentiate the pairs.

Reply:
Ok. I'm going to make an exception when what's returned is a non-templated string.

Reply:
That does seem reasonable, but then you can't assert that the "value" is a function and help prevent programmer error.